go/database/mpt/proof.go

```diff
@@ -0,0 +1,150 @@
+// If the witness proof contains an account for the input address, it returns its information.
+// If the proof does not contain an account, it returns false.
+// The method may return an error if the proof is invalid.
+func (p WitnessProof) GetAccountInfo(root common.Hash, address common.Address) (AccountInfo, bool, error) {
+	path := AddressToNibblePath(address, nil) // TODO hash path without the node source
+	node := p.getNext(root, path)
+	if node == nil {
+		return AccountInfo{}, false, nil
+	}
+	return node.(*AccountNode).info, true, nil
+}
+
+// GetBalance extracts a balance from the witness proof for the input root hash and the address.
+// If the witness proof contains the requested account for the input address for the given root hash, it returns its balance.
+// If the proof does not contain an account, it returns false.
+// The method may return an error if the proof is invalid.
+func (p WitnessProof) GetBalance(root common.Hash, address common.Address) (common.Balance, bool, error) {
+	return common.Balance{}, false, nil
+}
+
+// GetNonce extracts a nonce from the witness proof for the input root hash and the address.
+// If the witness proof contains an account for the input address, it returns its nonce.
+// If the proof does not contain an account, it returns false.
+// The method may return an error if the proof is invalid.
+func (p WitnessProof) GetNonce(root common.Hash, address common.Address) (common.Nonce, bool, error) {
+	return common.Nonce{}, false, nil
+}
+
+// GetCodeHash extracts a code hash from the witness proof for the input root hash and the address.
+// If the witness proof contains an account for the input address, it returns its code hash.
+// If the proof does not contain an account, it returns false.
+// The method may return an error if the proof is invalid.
+func (p WitnessProof) GetCodeHash(root common.Hash, address common.Address) (common.Hash, bool, error) {
+	return common.Hash{}, false, nil
+}
+
+// GetState extracts a storage slot from the witness proof for the input root hash, account address and the storage key.
+// If the witness proof contains an input storage slot for the input key, it returns its value.
+// If the proof does not contain a slot, it returns false.
+// The method may return an error if the proof is invalid.
+func (p WitnessProof) GetState(root common.Hash, address common.Address, key common.Key) (common.Value, bool, error) {
+	path := KeyToNibblePath(key, nil) // TODO hash path without the node source
+	node := p.getNext(root, path)
+	if node == nil {
+		return common.Value{}, false, nil
+	}
+	return node.(*ValueNode).value, true, nil
+}
+
+// AllStatesZero checks that all storage slots are empty for the input root hash, account address and the storage key range.
+// If the witness proof contains all empty slots for the input key range, it returns true.
+// An empty slot is a slot that contains a zero value, or does not exist at all.
```

Comment:
All slots `exist`. There is no such thing as a non-existing slot.

Reply:
ok, will remove